### PR TITLE
Section Render Refactor - History section

### LINKED ISSUE #1452: Section Render Refactor - History section
Make same changes as https://github.com/18F/e-QIP-prototype/pull/1431 to the History section, including:
- Create `Intro, Review` components
- Create `HistoryConnector` for passing props into subsection components
- Use `<Route>` components to render `<History>` and its subsections
- Use `@config/formSections` config instead of `@config/navigation`
- Each subsection should extend `Subsection` instead of `SubsectionElement`
- General cleanup, use classnames, import aliases, remove unused props, etc.

### PR DESCRIPTION
**Pull request by suzubara**
## Description
- Fixes #1452 
- Render `History` section and its subsections using `<Route>` instead of `<SectionViews>`
- Created standalone components for summary progress for Education/Employment/Residence
- Created wrapper components for Education/Employment/Residence/Federal to bridge rendering & functions previously handled by the `History` component
  - This definitely deserves further cleanup, I think maybe after we work on the Accordion component
- Created `HistoryConnector` HOC and tests for it (replacing some `History.test.jsx` tests)
- Consolidated some common functions into `History/helpers.js`
- Converted `overrideInitial` prop to bool instead of function in order to simplify it
- _Not done:_ dynamic `totalYears` value based on form type -- I think this should be a separate story

## Checklist for Reveiwer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

### CODE REVIEW

File: src/components/Section/History/Education/Education.jsx [hunk 1]
```diff
@@ -1,56 +1,79 @@
...
+import { HISTORY, HISTORY_EDUCATION } from 'config/formSections/history'
+
+import schema from 'schema'
+import validate, { EducationItemValidator } from 'validators'
+
+import Subsection from 'components/Section/shared/Subsection'
+import { Accordion } from 'components/Form'
+import { openState } from 'components/Form/Accordion/Accordion'
+
+import connectHistorySection from '../HistoryConnector'
```
Comment (gidjin) [inline src/components/Section/History/Education/Education.jsx]: Should we be adding relative imports? Is there a recommendation we can make like no more than one level up?

Comment (suzubara) [reply src/components/Section/History/Education/Education.jsx]: @gidjin Yeah I am fine with that

**Comment by suzubara** [comment]
@gidjin Hm, I'm not seeing that on my local.. what's your user's birthdate?

**Comment by gidjin** [comment]
You are right I had a birthday that was under 18 so the form correctly switched to 2 year minimum of history. Changing the birthday it went back to 10 years as expected.

**Pull request merged**